AIDevGallery/Controls/SampleContainer.xaml.cs

```diff
@@ -178,7 +180,9 @@ public async Task LoadSampleAsync(Sample? sample, List<ModelDetails>? models)
                     SampleFrame.Content = null;
 
                     VisualStateManager.GoToState(this, "WcrModelNeedsDownload", true);
-                    if (!await modelDownloader.SetDownloadOperation(ModelType.PhiSilica, sample.Id, LanguageModel.MakeAvailableAsync))
+                    if (!await modelDownloader.SetDownloadOperation(ModelType.PhiSilica, sample.Id, LanguageModel.MakeAvailableAsync) ||
```

Comment:
Bring the lines:
```cs
_sampleLoadingCts = new CancellationTokenSource();
var token = _sampleLoadingCts.Token;
```
to before the `// if PhiSilica, only show model loader and reload sample once loaded`

Then change this line to:
```suggestion
                    if (!await modelDownloader.SetDownloadOperation(ModelType.PhiSilica, sample.Id, LanguageModel.MakeAvailableAsync).WaitAsync(token) ||
```